nncf/quantization/algorithms/accuracy_control/ranker.py

```diff
@@ -98,8 +98,9 @@ def find_groups_of_quantizers_to_rank(self, quantized_model_graph: NNCFGraph) ->
             if x.metatype in self._algo_backend.get_quantizer_metatypes()
         ]
 
-        quantized_model_graph_without_shapeof = remove_shapeof_subgraphs(
-            deepcopy(quantized_model_graph), self._algo_backend.get_shapeof_metatypes()
+        quantized_model_graph_without_shapeof = deepcopy(quantized_model_graph)
+        remove_shapeof_subgraphs_inplace(
+            quantized_model_graph_without_shapeof, self._algo_backend.get_shapeof_metatypes()
```

Comment:
Why do we need to remove it in-place?

Reply:
The function was working inplace, I make it clear by naming them `..._inplace`. Should I refactor all this functions to be out of place?

Reply:
It's right, before, this method was doing its job in place. I suggest placing graph copying into this method and making it return the new `NNCFGraph` instance.

Reply:
It will affect performance as all steps during `get_inference_graph` will be coping models inside. Is that acceptable?

Reply:
I would suggest that you implement copying only once as it is done for now. Not for each step as default.

Reply:
This comment is outdated, `transform_to_inference_graph` is reverted back